lib/src/lifecycle_module.dart

```diff
@@ -77,12 +77,12 @@ abstract class LifecycleModule {
   Future loadChildModule(LifecycleModule newModule) async {
     newModule.didLoad.listen((_) {
       _childModules.add(newModule);
-      _didLoadChildModuleController.add(newModule);
     });
     newModule.willUnload.listen((_) {
       _childModules.remove(newModule);
     });
     await newModule.load();
+    await _didLoadChildModuleController.add(newModule);
```

Comment:
i moved this here because I couldn't successfully await this during the listener above